warehouse/query-core/src/main/java/datawave/query/jexl/functions/TermFrequencyAggregator.java

```diff
@@ -60,4 +68,45 @@ protected Key getSeekStartKey(Key current, ByteSequence pointer) {
         // seek to the next documents TF
         return new Key(current.getRow(), current.getColumnFamily(), new Text(pointer + Constants.NULL_BYTE_STRING + Constants.MAX_UNICODE_STRING));
     }
+    
+    @Override
+    public Key apply(SortedKeyValueIterator<Key,Value> itr, Document doc, AttributeFactory attrs) throws IOException {
+        Key key = super.apply(itr, doc, attrs);
+        
+        // only return a key if something was added to the document, documents that only contain Document.DOCKEY_FIELD_NAME as they found nothing of value to
+        // aggregate
+        if (doc.size() == 1 && doc.get(Document.DOCKEY_FIELD_NAME) != null) {
+            key = null;
+            
+            // empty the document
+            doc.remove(Document.DOCKEY_FIELD_NAME);
+        }
+        
+        return key;
+    }
+    
+    /**
+     * Limit keep fields to those that are index only or if there is no filter specified
+     * 
+     * @param topKey
+     * @param fieldNameValue
+     * @return
+     */
+    @Override
+    protected boolean toKeep(Key topKey, Tuple2<String,String> fieldNameValue) {
+        return fieldsToKeep == null || filter == null || fieldsToKeep.contains(JexlASTHelper.removeGroupingContext(fieldNameValue.first()));
```

Comment:
Since the filter is only being used for the addToDoc concept, is it even to be considered here?  As you have it written here, If we have fieldsToKeep specified but no filter then we keep all fields?

Reply:
will remove filter == null as a condition